controllers/v1/assessments.js

```diff
@@ -835,3 +835,148 @@ async function getUserProfileFunc(req,res){
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+* "entityId": "",
+* "entityType": "",
+* "programId": "",
+* "solutionId": ""
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*   {
+    "result": true,
+    "programName": "",
+    "solutionName": "",
+    "data": [{
+        "chart": {
+            "type": "bar"
+        },
+        "title": {
+            "text": ""
+        },
+        "xAxis": [
+            {
+                "categories": []
+            },
+            {
+                "opposite": true,
+                "reversed": false,
+                "categories": [],
+                "linkedTo": 0
+            }
+        ],
+        "yAxis": {
+            "min": 0,
+            "title": {
+                "text": ""
+            }
+        },
+        "legend": {
+            "reversed": true
+        },
+        "plotOptions": {
+            "series": {
+                "stacking": "percent"
+            }
+        },
+        "series": [
+            {
+                "name": "Level 1",
+                "data": []
+            }
+        ]
+    },
+    {
+        "domainName": "",
+        "domainExternalId": "",
+        "criterias": [{
+             "criteriaName": "",
+             "levels": []
+        }]
+    }]
+}
+   * @apiUse errorBody
+   */
+exports.entityReport = async function(req,res){
+
+    let data = await entityReportChartCreateFunction(req, res);
+    res.send(data);
+}
+
+
+async function entityReportChartCreateFunction(req, res) {
+return new Promise(async function (resolve, reject) {
+
+    if (!req.body.entityId || !req.body.entityType || !req.body.programId || !req.body.solutionId) {
+        let response = {
+            result: false,
+            message: 'entityId,entityType,programId,solutionId are required fields'
+        }
+        resolve(response);
+    }
+    else {
+           //get domainName and level info
+            model.MyModel.findOneAsync({ qid: "entity_level_assessment_report_query" }, { allow_filtering: true })
+                .then(async function (result) {
+
+                    var bodyParam = JSON.parse(result.query);
+                    if (config.druid.assessment_datasource_name) {
+                        bodyParam.dataSource = config.druid.assessment_datasource_name;
+                    }
+
+                    //dynamically appending values to filter
+                    bodyParam.filter.fields[0].dimension = req.body.entityType;
+                    bodyParam.filter.fields[0].value = req.body.entityId;
+                    bodyParam.filter.fields[1].value = req.body.programId;
+                    bodyParam.filter.fields[2].value = req.body.solutionId;
+
+                    let options = config.druid.options;
+                    options.method = "POST";
+                    options.body = bodyParam;
+
+                    let data = await rp(options);
```

Comment:
@deepakr123 - Use a limit to load only the threshold no of documents in descending order by started date

Reply:
resolved